torch/_dynamo/variables/functions.py

```diff
@@ -289,7 +306,7 @@ def bind_args(self, parent, args, kwargs):
                         contents_var,
                     )
 
-                result[name] = out
+                closure_cells[name] = out
```

Comment:
This is the critical change, everything else is documentation or removing the now dead code paths.